### ISSUE TITLE
Speculative-T completion provider does not trigger for generic parameter in async return type

### ISSUE THREAD

**Issue by agocke** [dotnet/roslyn#14525]
This test demonstrates the problem:

```csharp
        [Fact]
        [Test.Utilities.CompilerTrait(Test.Utilities.CompilerFeature.LocalFunctions)]
        public async Task LocalFunctionAfterAyncTask()
        {
            var markup = @"
class C
{
    public void M()
    {
        async Task<$$>
    }
}";
            await VerifyItemExistsAsync(markup, "T");
        }
```

The underlying problem here is that local functions don't have the parsing logic to parse `async Task<$$>` as a local function with an `async` modifier and a `Task<>` return type -- instead this is an `async` return type and `Task` identifier.

The logic for members is not as simple for local functions and will require a bit of refactoring.
